### ISSUE TITLE
*args and **kwargs required for ParamSpec not accepting them

### ISSUE THREAD

**Issue by ethantkoenig** [astral-sh/ty#2926]
### Summary

Given the following code:

```python
from collections.abc import Callable
from typing import Generic, ParamSpec, Protocol

P = ParamSpec('P')


def constructor(type_: 'type[HasConstructor[P]]') -> Callable[P, object]:
    return type_


class HasConstructor(Generic[P], Protocol):
    def __init__(self, *args: P.args, **kwargs: P.kwargs): ... 


constructor(object)()
```

ty emits the following error:

```
error[missing-argument]: No arguments provided for required parameters `*args`, `**kwargs`
  --> example.py:15:1
   |
15 | constructor(object)()
   | ^^^^^^^^^^^^^^^^^^^^^
   |
info: These arguments are required because `ParamSpec` `P'return` could represent any set of parameters at runtime
info: rule `missing-argument` is enabled by default

Found 1 diagnostic
```

But the code should pass type checking, because `object.__init__` doesn't accept any positional or keyword arguments.

https://play.ty.dev/638af0f4-42ef-46b2-8aef-51abd885e4e1

### Version

ty 0.0.18

**Comment by dhruvmanila** [comment]
I think this requires https://github.com/astral-sh/ty/issues/903 first.
